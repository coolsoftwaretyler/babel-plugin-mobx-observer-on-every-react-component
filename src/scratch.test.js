import { expect, test, describe } from "bun:test";
import { transform } from "@babel/core";
import pluginSyntaxJsx from "@babel/plugin-syntax-jsx";
import pluginTransReactJsx from "@babel/plugin-transform-react-jsx";
import { autoObserverPlugin } from "./index.mjs";

const runTransform = (input) => {
  return transform(input, {
    plugins: [pluginSyntaxJsx, pluginTransReactJsx, autoObserverPlugin],
  });
};

describe("Mobx Observer Babel Plugin", () => {
  test("specific file", () => {
    const input = `
import * as React from 'react'
import { Screen } from '@components/common/Screen'
import hideTabBar from '@hooks/hideTabBar'
import UniversalToolHeader from '@components/common/UniversalToolHeader/UniversalToolHeader'
import WalterPicksErrorBoundary from '@components/common/WalterPicksErrorBoundary'
import { useNavigation } from '@react-navigation/native'
import WalterText from '@components/common/WalterText'
import { RebopConsole } from '@rebop/react-native-rebop-console'
import { View } from 'react-native'

export function PreviewDeployConsoleScreen() {
  hideTabBar()
  const navigation = useNavigation()

  const projectId = 'walterpicks-mobile-app'

  return (
    <Screen preset="fixed" safeAreaEdges={['top']} testID="PreviewDeployConsoleScreenScreen">
      <UniversalToolHeader title="Preview Deploys" />
      <WalterPicksErrorBoundary navigation={navigation}>
        <View
          style={{
            flexDirection: 'column',
            justifyContent: 'center',
            alignItems: 'center',
          }}
        >
          <WalterText style={{ marginLeft: 8 }}>sentinel value: Test 413-Brierwood</WalterText>
        </View>
        <RebopConsole projectId={projectId} />
      </WalterPicksErrorBoundary>
    </Screen>
  )
}

        
        `;

    const out = runTransform(input);

    expect(out.code).toMatchSnapshot();
  });
});
