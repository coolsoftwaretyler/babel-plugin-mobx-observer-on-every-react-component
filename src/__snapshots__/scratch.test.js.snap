// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobx Observer Babel Plugin specific file 1`] = `
"import { observer } from "mobx-react";
import * as React from 'react';
import { Screen } from '@components/common/Screen';
import hideTabBar from '@hooks/hideTabBar';
import UniversalToolHeader from '@components/common/UniversalToolHeader/UniversalToolHeader';
import WalterPicksErrorBoundary from '@components/common/WalterPicksErrorBoundary';
import { useNavigation } from '@react-navigation/native';
import WalterText from '@components/common/WalterText';
import { RebopConsole } from '@rebop/react-native-rebop-console';
import { View } from 'react-native';
export function PreviewDeployConsoleScreen() {
  return observer(function PreviewDeployConsoleScreen() {
    hideTabBar();
    const navigation = useNavigation();
    const projectId = 'walterpicks-mobile-app';
    return /*#__PURE__*/React.createElement(Screen, {
      preset: "fixed",
      safeAreaEdges: ['top'],
      testID: "PreviewDeployConsoleScreenScreen"
    }, /*#__PURE__*/React.createElement(UniversalToolHeader, {
      title: "Preview Deploys"
    }), /*#__PURE__*/React.createElement(WalterPicksErrorBoundary, {
      navigation: navigation
    }, /*#__PURE__*/React.createElement(View, {
      style: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center'
      }
    }, /*#__PURE__*/React.createElement(WalterText, {
      style: {
        marginLeft: 8
      }
    }, "sentinel value: Test 413-Brierwood")), /*#__PURE__*/React.createElement(RebopConsole, {
      projectId: projectId
    })));
  });
}"
`;
